src/main/java/org/folio/rest/transaction/PaymentCreditHandler.java

```diff
@@ -0,0 +1,387 @@
+package org.folio.rest.transaction;
+
+import static java.util.stream.Collectors.groupingBy;
+import static java.util.stream.Collectors.toMap;
+import static org.apache.commons.collections4.CollectionUtils.isNotEmpty;
+import static org.folio.rest.impl.BudgetAPI.BUDGET_TABLE;
+import static org.folio.rest.impl.TransactionSummaryAPI.INVOICE_TRANSACTION_SUMMARIES;
+import static org.folio.rest.persist.HelperUtils.getCriterionByFieldNameAndValueNotJsonb;
+import static org.folio.rest.persist.HelperUtils.getFullTableName;
+import static org.folio.rest.persist.HelperUtils.handleFailure;
+
+import io.vertx.core.AsyncResult;
+import io.vertx.core.Context;
+import io.vertx.core.Future;
+import io.vertx.core.Handler;
+import io.vertx.core.Promise;
+import io.vertx.core.json.JsonArray;
+import io.vertx.core.json.JsonObject;
+import io.vertx.ext.web.handler.impl.HttpStatusException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.function.Function;
+import java.util.function.Predicate;
+import java.util.stream.Collectors;
+import javax.money.CurrencyUnit;
+import javax.money.Monetary;
+import javax.ws.rs.core.Response;
+import org.apache.commons.lang3.StringUtils;
+import org.folio.rest.jaxrs.model.Budget;
+import org.folio.rest.jaxrs.model.Error;
+import org.folio.rest.jaxrs.model.Errors;
+import org.folio.rest.jaxrs.model.Transaction;
+import org.folio.rest.jaxrs.resource.FinanceStorageTransactions;
+import org.folio.rest.persist.MoneyUtils;
+import org.folio.rest.persist.Tx;
+import org.folio.rest.persist.Criteria.Criterion;
+
+public class PaymentCreditHandler extends AllOrNothingHandler {
+
+  private static final String TEMPORARY_INVOICE_TRANSACTIONS = "temporary_invoice_transactions";
+  private static final String TRANSACTIONS_TABLE = "transaction";
+
+  Predicate<Transaction> hasEncumbrance = txn -> txn.getPaymentEncumbranceId() != null;
+  Predicate<Transaction> isPaymentTransaction = txn -> txn.getTransactionType().equals(Transaction.TransactionType.PAYMENT);
+  Predicate<Transaction> isCreditTransaction = txn -> txn.getTransactionType().equals(Transaction.TransactionType.CREDIT);
+
+  public PaymentCreditHandler(Map<String, String> okapiHeaders, Context ctx, Handler<AsyncResult<Response>> asyncResultHandler) {
+    super(TEMPORARY_INVOICE_TRANSACTIONS, INVOICE_TRANSACTION_SUMMARIES, okapiHeaders, ctx, asyncResultHandler);
+  }
+
+  @Override
+  String getSummaryId(Transaction transaction) {
+    return transaction.getSourceInvoiceId();
+  }
+
+  @Override
+  Criterion getTransactionBySummaryIdCriterion(String value) {
+    return getCriterionByFieldNameAndValueNotJsonb("sourceInvoiceId", "=", value);
+  }
+
+  @Override
+  public void createTransaction(Transaction transaction) {
+    processAllOrNothing(transaction, this::processAllPaymentsCredits).setHandler(result -> {
```

Comment:
It is possible not to override this method, but instead to override `AllOrNothingHandler#createPermanentTransactions` ( and maybe give it more appropriate name).

Reply:
Can you elaborate a little?

Reply:
If `PaymentCreditHandler#createPermanentTransactions` is a overriding of the AllOrNothingHandler#createPermanentTransactions , so child-class's method will be called https://github.com/folio-org/mod-finance-storage/blob/ec334af25cb191714e610edaa99822851213a6dc/src/main/java/org/folio/rest/transaction/AllOrNothingHandler.java#L54.